File: sortingAlgos(java)/src/sorting/insertionSort.java
package sorting;

import java.util.Scanner;

public class insertionSort {

	public static void main(String[] args) {
		Scanner cin = new Scanner(System.in);
		
		int[] arr = new int[100];
		int n, j;

		//input
		n = cin.nextInt();
		for(int i = 0; i < n; i++) arr[i] = cin.nextInt();
		
		for(int i = 1; i < n; i++) {
			j = i;
			while(j > 0 && arr[j-1] > arr[j]) {
				int tmp = arr[j-1];
				arr[j-1] = arr[j];
				arr[j] = tmp;
				j--;
			}
		}
		
		//printing
		for(int i = 0;i < n; i++) System.out.print(arr[i] + "  ");
	}

}

File: sortingAlgos/src/insertionSort.cpp
/*
#include<bits/stdc++.h>
using namespace std;

int main() {
	int n, arr[100], j;
	cin >> n;

	//input
    for(int i = 0; i < n; i++) cin >> arr[i];

    for(int i = 1; i < n; i++) {
    	j = i;
    	while(j > 0 && arr[j-1] > arr[j]) {
    		swap(arr[j-1], arr[j]);
    		j--;
    	}
    }

	for(int i = 0;i < n; i++) cout << arr[i] << " ";

	return 0;
}
*/
